### ISSUE TITLE
Cannot suppress `TrailingCommaOnDeclarationSite ` on enum with trailing semicolon

### ISSUE THREAD

**Issue by alvr** [detekt/detekt#5566]
## Expected Behavior
When using `TrailingCommaOnDeclarationSite`, if an enum has a semicolon to separate the last item, KtLint adds a comma before. I reported the error in pinterest/ktlint/issues/1714 and it seems that the KtLint suppress is working, but the suppress for detekt is not.

```kotlin
enum class Example {
    A,
    B;

    fun something() = "blah"
}

enum class Example2 {
    A,
    B;

    companion object {
        val values = listOf(A, B)
    }
}
```

is formatted into

```kotlin
enum class Example {
    A,
    B,
    ;

    fun something() = "blah"
}

enum class Example2 {
    A,
    B,
    ;

    companion object {
        val values = listOf(A, B)
    }
}
```

## Observed Behavior
If I add `@Suppress("TrailingCommaOnDeclarationSite")` before the declaration of the enum, the error won't be suppressed and is still formatting the enum.

## Steps to Reproduce
Using this configuration of detekt

```yaml
TrailingCommaOnCallSite:
  active: true
  autoCorrect: true
  useTrailingCommaOnCallSite: true
TrailingCommaOnDeclarationSite:
  active: true
  autoCorrect: true
  useTrailingCommaOnDeclarationSite: true
```

and

```kotlin
@Suppress("TrailingCommaOnDeclarationSite")
enum class Example {
    A,
    B;

    fun something() = "blah"
}
```

## Context
This happens with detekt 1.22.0. The error should be suppressed, otherwise it will format the enum adding a trailing comma before the semicolon.

**Comment by cortinico** [comment]
> it seems that the KtLint suppress is working, but the suppress for detekt is not.

I believe this might be related on how suppression works for the `formatting` ruleset. It should work if you suppress on the whole file though (still not ideal).

**Comment by Bryan-Lamb** [comment]
Also experiencing this issue 👍 , would love some additional granularity for configuration here
